### ISSUE TITLE
`getByRole` Support of `summary` Elements

### ISSUE THREAD

**Issue by ITenthusiasm** [testing-library/dom-testing-library#1252]
### Desired Feature

The [`<summary>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary) -- together with the [`<details>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) element -- plays an incredibly important role in supporting "Accordion Components" natively -- and without the need for JS! They _are_ exposed to the accessibility tree and discoverable for Screen Readers (tested with Voice Over in FireFox/Safari/Chrome on MacOS). However, this important element is not currently discoverable for developers using the recommended `getByRole` query.

If possible, it would be great if `getByRole` could support `<summary>`.

### Suggested Implementation:

According to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/summary#technical_summary), the implicit ARIA role is `button`. So a simple solution could be to allow `summary` elements to be discoverable by `getByRole("button")`. However, one thing to keep in mind is that according to the [spec](https://w3c.github.io/html-aria/#el-summary):

> Many, but not all, user agents expose the summary element with an implicit ARIA role=[button](https://w3c.github.io/html-aria/#index-aria-button).

When I tested with VoiceOver, the `summary` element seemed to be identified as a unique kind of button that was related to the `group` created by the `details` element. Based on other people's [previous experiences](https://accessibleculture.org/articles/2012/03/screen-readers-and-details-summary/), other Screen Readers seem to do something similar. So the hope is that this would be a reasonable change.

### Alternatives

Another alternative is to do nothing and hope that the spec comes up with a true implicit role that isn't dictated by User Agents like popular Screen Readers. However, it's not clear how long that would take  (or if that would ever happen).

### Teachability, Documentation, Adoption, Migration Strategy:

The documentation probably would not need an update. More than likely, the Changelog could just mention that `summary` elements are newly discoverable by using `getByRole`.

**Comment by cmorten** [comment]
R.E.

> Another alternative is to do nothing and hope that the spec comes up with a true implicit role

See https://github.com/w3c/html-aam/issues/345#issuecomment-1654013809. _Feels_ like there is some traction in this space.

**Comment by ITenthusiasm** [comment]
Oh wow. That's a really helpful link. Yeah it looks like it wouldn't be so easy to move forward with something like this then. I guess in the meantime we'd need to stick to `getByText("SUMMARY_TEXT", { selector: "details > summary" })`?

**Comment by MatanBobi** [comment]
Hi @ITenthusiasm! Thanks for taking the time to open this issue.
This is going to be a tough one. We're using `aria-query` for the implicit role mappings and they follow the spec only. We don't want to have code which is not spec compliant because it will open up a new dependency for us, so I guess that this is better opened in `[aria-query](https://github.com/A11yance/aria-query)`.

**Comment by ITenthusiasm** [comment]
@MatanBobi Makes sense! Thanks for reviewing the issue and providing that background! 🙏🏾  It makes sense to stay close to the spec and avoid extra dependencies. I'll reach out to them soon.

**Comment by ITenthusiasm** [comment]
@MatanBobi Created A11yance/aria-query#546. Leaving a note here as a reference. @cmorten's link is also worth keeping track of.

**Comment by weilbith** [comment]
I've a similar issue, but for me it starts with querying for the `<details>` element already. According to the [specification](https://www.w3.org/TR/html-aria/#docconformance) this has an **explicit** role of `group`. As a `<details>` element can also have the `name` attribute, I thought this query should work: `getByRole("group", { name: "my-name" })`. But it doesn't (also without the `name`).
So I guess this should actually work. Shouldn't it?

**Comment by weilbith** [comment]
Thank you very much for the quick answer!
You're absolutely right. I mixed that up, sorry. 🙈 
So I guess I'm stuck on the same issue to properly select the `summary` of the `details`. 😕

**Comment by beanscg** [comment]
Bean Labs took a public-source pass on this issue.

The narrowest useful next step looks like: Offer a public comment that narrows the keyboard/ARIA or screen-reader expectation and identifies the component, example, or docs page to verify.

Public evidence checked:
- GitHub direct issue read: open, unassigned, comments=10, labels=enhancement, aria-query.
- Repository metadata read: license=MIT; archived=false.
- Body references files/examples: CODE_OF_CONDUCT.md

A good acceptance check would be a narrow public change that directly exercises the issue's reported boundary.

No PR opened from this pass.
